sdk/core/azure-core/test/ut/datetime_test.cpp

```diff
@@ -902,3 +902,14 @@ TEST(DateTime, Rfc3339Space)
       = DateTime::Parse("2022-08-24 00:43:08.0004308Z", DateTime::DateFormat::Rfc3339);
   EXPECT_EQ(datetime.ToString(DateTime::DateFormat::Rfc3339), "2022-08-24T00:43:08.0004308Z");
 }
+
+TEST(DateTime, DateOnly)
+{
+  {
+    auto dt = DateTime::Parse("1975-04-04T01:23:45.6789Z", DateTime::DateFormat::Rfc3339);
+
+    EXPECT_EQ(
+        dt.ToString(DateTime::DateFormat::Rfc3339, DateTime::TimeFractionFormat::Date),
+        "1975-04-04");
```

Comment:
Would this be considered data loss when round-tripping @LarryOsterman? Should we only ever return `DateOnly` if the time is exactly 0?

Reply:
I don't believe it's possible to round-trip this. The date-only format is an invalid Rfc3339 date-time string and we parse date-time strings, not date-only strings.